src/main/java/com/job_search/controller/CompanyController.java

```diff
@@ -0,0 +1,75 @@
+package com.job_search.controller;
+
+import com.job_search.controller.dto.CompanyDto;
+import com.job_search.service.CompanyService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+import static java.lang.String.format;
+
+
+@RestController
+@RequestMapping("/api/companies")
+public class CompanyController {
+
+    @Autowired
+    private CompanyService companyService;
+
+    @GetMapping
+    public List<CompanyDto> getAllCompanies() {
+        return companyService.getAllCompanies();
+    }
+
+    @GetMapping("/company")
+    public CompanyDto getCompanyByName(@RequestParam String name) {
+        return companyService.getCompanyByName(name);
+    }
+
+    @PostMapping
+    public ResponseEntity createCompany(@RequestBody CompanyDto companyDto) {
+        if (companyService.existsByName(companyDto.getName()))
+            return ResponseEntity
+                    .status(HttpStatus.BAD_REQUEST)
+                    .body(format("The '%s' company already exist", companyDto.getName()));
+        companyService.createCompany(companyDto);
+        return ResponseEntity
```

Comment:
Based on common practices of the REST architecture, return created entity. Additionally, I'm suggesting wrapping all your objects into ResponseEntity<> it will simplify the development of the integration tests in the next task

Reply:
Thanks, all commets were fixed in https://github.com/Awesome5040/job-search/pull/1/commits/d2f6024a4fdf4c4e2837e061af5d95193c18c0b0